Add OpenGraph tags for Article content

patrickfournier opened the pull request.

patrickfournier committed:
Generate OpenGraph tags for Article content type.

Additionally, adds a setting for mapping author names to X (Twitter) accounts, to be consistent with Facebook profiles.

patrickfournier commented:
This should be merged after PR #78.

MaevaBrunelles commented:
I let you rebase and mark the PR as ready for review when it's good for you.

patrickfournier committed:
Add test and documentation.

justinmayer reviewed (approved):
Nice work, Patrick 👍

MaevaBrunelles commented on README.md:
```diff
@@ -61,6 +61,26 @@ Please note that :
  * `SEO_ENHANCER_OPEN_GRAPH` setting must be enabled in order to use
   `SEO_ENHANCER_TWITTER_CARDS` (see [why](#twitter-cards))
 
+When Open Graph and Twitter Cards features are enabled, the SEO Enhancer will try to add
```
Details about how the feature works should be included in [this section](https://github.com/pelican-plugins/seo/blob/main/README.md#twitter-cards) (or in a new section just after the twitter cards), but not here as it's the global explanation of the plugin.

MaevaBrunelles commented on pelican/plugins/seo/seo_enhancer/html_enhancer/__init__.py:
```diff
@@ -88,7 +88,25 @@ def __init__(self, file, output_path, path, open_graph=False, twitter_cards=Fals
+                    modified=_modified,
+                    category=_category.name,
+                    tags=getattr(file, "tags", []),
+                    author=_metadata.get("fb_profile") or _author_profile,
```
Same here, I think we should have two tests (or a parametrized one) [here](https://github.com/pelican-plugins/seo/blob/main/pelican/plugins/seo/tests/test_seo_enhancer.py) to ensure that the open graph element is filled by both `SEO_ENHANCER_AUTHOR_FACEBOOK_PROFILES` and `SEO_ENHANCER_AUTHOR_TWITTER_PROFILES` new settings if filled, or is filled by using fallback to `_author_profile`.

patrickfournier committed:
Improve documentation (address PR comment)

patrickfournier committed:
Add tests

- Check for article OG tags in generated HTML.
- Check author tag is properly set either from article metadata or site settings.

MaevaBrunelles commented:
All good for me, the PR is merged. Thanks again for your contribution @patrickfournier 💯
